scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala

```diff
@@ -289,158 +298,154 @@ trait ClassLikeSupport:
     def isTreeAbstract(typ: Tree): Boolean = typ match {
       case TypeBoundsTree(_, _) => true
       case LambdaTypeTree(params, body) => isTreeAbstract(body)
       case _ => false
     }
 
-
     val (generics, tpeTree) = typeDef.rhs match
-      case LambdaTypeTree(params, body) => (params.map(parseTypeArgument), body)
+      case LambdaTypeTree(params, body) => (params.map(mkTypeArgument), body)
       case tpe => (Nil, tpe)
 
-    new DProperty(
-      typeDef.symbol.dri,
-      typeDef.symbol.normalizedName,
-      /*documentation =*/ typeDef.symbol.documentation.asJava,
-      /*expectPresentInSet =*/ null, // unused
-      /*sources =*/ JMap(),
-      /*visibility =*/ placeholderVisibility,
-      /*type =*/ tpeTree.dokkaType, // TODO this may be hard...
-      /*receiver =*/ null, // Not used
-      /*setter =*/ null,
-      /*getter =*/ null,
-      /*modifier =*/ placeholderModifier,
-      ctx.sourceSet.toSet,
-      /*generics =*/ generics.asJava, // TODO
-       /*isExpectActual =*/ false,
-      PropertyContainer.Companion.empty() plus MemberExtension(
+    mkMember(
+      typeDef.symbol,
+      MemberExtension(
         typeDef.symbol.getVisibility(),
         typeDef.symbol.getExtraModifiers(),
-        Kind.Type(!isTreeAbstract(typeDef.rhs), typeDef.symbol.isOpaque),
+        Kind.Type(!isTreeAbstract(typeDef.rhs), typeDef.symbol.isOpaque, generics),
         typeDef.symbol.getAnnotations(),
         tpeTree.dokkaType.asSignature,
         typeDef.symbol.source
         )
     )
 
-  def parseValDef(valDef: ValDef): DProperty =
+  def parseValDef(valDef: ValDef): Member =
     def defaultKind = if valDef.symbol.flags.is(Flags.Mutable) then Kind.Var else Kind.Val
     val kind = if valDef.symbol.flags.is(Flags.Implicit) then
         Kind.Implicit(Kind.Val, extractImplicitConversion(valDef.tpt.tpe))
         else defaultKind
 
-    new DProperty(
-      valDef.symbol.dri,
-      valDef.symbol.normalizedName,
-      /*documentation =*/ valDef.symbol.documentation.asJava,
-      /*expectPresentInSet =*/ null, // unused
-      /*sources =*/ JMap(),
-      /*visibility =*/ placeholderVisibility,
-      /*type =*/ valDef.tpt.dokkaType,
-      /*receiver =*/ null, // Not used
-      /*setter =*/ null,
-      /*getter =*/ null,
-      /*modifier =*/ placeholderModifier,
-      ctx.sourceSet.toSet,
-      /*generics =*/ JList(),
-       /*isExpectActual =*/ false,
-      PropertyContainer.Companion.empty().plus(MemberExtension(
+    mkMember(
+      valDef.symbol,
+      MemberExtension(
           valDef.symbol.getVisibility(),
           valDef.symbol.getExtraModifiers(),
           kind,
           valDef.symbol.getAnnotations(),
           valDef.tpt.tpe.dokkaType.asSignature,
           valDef.symbol.source
-      ))
+      )
     )
+
+  def mkMember[T <: Kind](
+    symbol: Symbol,
+    member: MemberExtension,
+    compositeExt: CompositeMemberExtension = CompositeMemberExtension.empty): Member =
+      new DClass(
```

Comment:
I know we will discuss whether dokka will still remain as a base for our doctool, but if dokka stays, we should consider making `Member` a subtype od `Documentable`, and get rid of these hacks with `DClass` wrapper :P

Reply:
Actually using DClass is really conviniet :)